core/dbt/contracts/project.py

```diff
@@ -344,6 +344,7 @@ class ProjectFlags(ExtensibleDbtClassMixin):
     skip_nodes_if_on_run_start_fails: bool = False
     state_modified_compare_more_unrendered_values: bool = False
     state_modified_compare_vars: bool = False
+    allow_mf_time_spines_without_yaml_configuration: bool = False
```

Comment:
What do you think about the naming, `require_yaml_configuration_for_mf_time_spines: False` instead?

It would be a more conventional nam and better follow our guides: https://github.com/dbt-labs/dbt-core/blob/main/docs/guides/behavior-change-flags.md#rules-for-introducing-a-new-flag